src/pages/index.js

```diff
@@ -56,23 +60,169 @@ const cardSection = new Section(
   selectors.cardList
 );
 
-const createCard = (cardData) => {
-  const card = new Card(
-    {
-      data: cardData,
-      handleImageClick: () => {
-        popupWithImage.open(cardData);
-      },
-    },
-    selectors.cardTemplate
-  );
-  return card.generateCard();
-};
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "95813e96-588f-4415-8ffb-bcee955e5070",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      about: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => console.error(err));
+
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error("Error fetching initial cards:", err);
+  });
 
 userInfoPopup.setEventListeners();
 newCardPopUp.setEventListeners();
 popupWithImage.setEventListeners();
-cardSection.renderItems(initialCards);
+confirmPopup.setEventListeners();
+profileImagePopup.setEventListeners();
+
+/* Functions */
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    selectors.cardTemplate,
+    handleImageClick,
+    handleDeleteCard,
+    handleCardLike
+  );
+  return card.generateCard();
+}
+
+function handleImageClick(cardData) {
+  popupWithImage.open(cardData);
+}
+
+function handleDeleteCard(card) {
+  confirmPopup.setSubmitFunction(() => {
+    confirmPopup.setIsDeleting(true);
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.getElement().remove();
+        confirmPopup.setIsDeleting(false);
+      })
+      .catch((err) => {
+        console.error("Error deleting card:", err);
+      })
+      .finally(() => {
+        confirmPopup.close();
```

Comment:
`confirmPopup.close();` should be in `.then` in line 121


You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**